### PR TITLE
Events for conversations

### PR DESCRIPTION
**Pull request by anipetrov**
Adding to events for the conversations tab, `focusedOnConversationMessageInput` and `sentConversationMessage`!

### CODE REVIEW

File: src/Schema/Events/Conversations.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ *    action: "sentConversationMessage",
+ *    impulse_conversation_id: 198,
+ *    impulse_message_id: 8765
+ *    
+ *  }
+ * ```
+ */
+export interface SentConversationMessage {
+    action: ActionType.sentConversationMessage
+    impulse_conversation_id?: number
```
Comment (eessex) [inline src/Schema/Events/Conversations.ts:44]: Is there an instance where a conversation or message does not have an `id`?

Comment (anipetrov) [reply src/Schema/Events/Conversations.ts:44]: it should actually always have it, so i can remove the `?`!
also, i am resolving the conflicts but getting a yarn error, anyone available for a quick pair?

Comment (anipetrov) [reply src/Schema/Events/Conversations.ts]: nvm fixed it!

Comment (anipetrov) [reply src/Schema/Events/Conversations.ts]: i do also wonder if `number` is the right type given that these are postgresql ids?

File: src/Schema/Events/Conversations.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ *  {
+ *    action: "focusedOnConversationMessageInput",
+ *    impulse_conversation_id: 198
+ *
+ *  }
+ * ```
+ */
+export interface FocusedOnConversationMessageInput {
+  action: ActionType.focusedOnConversationMessageInput
+  impulse_conversation_id?: number
```
Comment (starsirius) [inline src/Schema/Events/Conversations.ts:24]: @anipetrov Do we also need a new context module or it's fine not supplying that for now?

File: src/Schema/Events/Conversations.ts (hunk 1 above)
Comment (eessex) [reply src/Schema/Events/Conversations.ts]: If redshift receives them as `number` then this type makes sense. And yes, the `?` being removed will make the property required.

File: src/Schema/index.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 export * from "./Events"
 export * from "./Events/Authentication"
+export * from "./Events/Conversations"
 export * from "./Events/Click"
```
Comment (starsirius) [inline src/Schema/index.ts:4]: @eessex A organization questions for events, looks like we have both actions (e.g. Click, Tap, etc.) and functionalities (e.g. Authentication, Conversations, etc.). Curious about guidelines for organizing new events.

Comment (eessex) [reply src/Schema/index.ts:4]: I think conversations makes sense if we expect more events to exist in this domain.

File: src/Schema/Events/Conversations.ts (hunk 2 above)
Comment (anipetrov) [reply src/Schema/Events/Conversations.ts:24]: no i dont think we need one!

**Comment by starsirius** [comment]
Thanks @anipetrov and @eessex for the review!

**Comment by artsyit** [comment]
:rocket: PR was released in [`v1.14.0`](https://github.com/artsy/cohesion/releases/tag/v1.14.0) :rocket:

File: src/Schema/Events/Conversations.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ *    impulse_conversation_id: 198,
+ *    impulse_message_id: 8765
+ *
+ *  }
+ * ```
+ */
+export interface SentConversationMessage {
+  action: ActionType.sentConversationMessage
+  impulse_conversation_id?: number
+  impulse_message_id?: number
```
Comment (eessex) [inline src/Schema/Events/Conversations.ts:45]: It looks like these properties are still optional, but should be required.

Comment (anipetrov) [reply src/Schema/Events/Conversations.ts:45]: ohhh i thought i pushed a commit to fix, will follow up

**Pull request merged**